### PR TITLE
Fix incrementing and decrementing beyond max cap

### PR DESCRIPTION
**Pull request by hanswilw**
Follow up on #830 

Fixes an error that occurs when no state exist in form as well.

### CODE REVIEW

File: app/reducers/events.js [hunk 1]
```diff
@@ -68,14 +68,16 @@ function mutateEvent(state: any, action: any) {
           [eventId]: {
             ...stateEvent,
             loading: false,
-            registrationCount: stateEvent.registrationCount + 1,
+            registrationCount: registration.pool
+              ? stateEvent.registrationCount + 1
+              : stateEvent.registrationCount,
```
Comment (ekmartin) [inline app/reducers/events.js]: why would the pool not exist if registrationCount exists?

Comment (hanswilw) [reply app/reducers/events.js]: When it is a socket registration to waiting list – the pool is not set.

File: app/reducers/forms.js [hunk 2]
```diff
@@ -22,6 +22,7 @@ export default formReducer.plugin({
         };
       case Event.SOCKET_UNREGISTRATION.SUCCESS:
       case Event.SOCKET_REGISTRATION.SUCCESS: {
+        if (!state) return undefined;
```
Comment (ekmartin) [inline app/reducers/forms.js]: `if (!state) return` should be the same if flow doesn't complain

Comment (hanswilw) [reply app/reducers/forms.js]: Didnt complain, but better off writing it like that. Fixed.

File: app/reducers/events.js (hunk 1 above)
Comment (ekmartin) [reply app/reducers/events.js]: alright

**Comment by hanswilw** [comment]
Agreed, I'll create an issue for it.

**Pull request merged**